include/dsn/tool/latency_tracer.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#pragma once
+#include <iterator>
+#include <string>
+#include <map>
+#include <dsn/service_api_c.h>
+#include <dsn/dist/fmt_logging.h>
+
+namespace dsn {
+namespace tool {
+
+struct latency_tracer
+{
+public:
+    uint64_t id;
+    std::string name;
+    uint64_t start_time;
+    uint64_t previous_time;
+    uint64_t end_time;
+    // it may be "get", "set" etc. you can set reasonable value to distinguish the requet type, it
+    // is initted "default" string
+    std::string request_type;
+    // record the the time used between current point and start point
+    std::unordered_map<std::string, uint64_t> points;
+    // record the the time used between current point and previous point
+    std::unordered_map<std::string, uint64_t> key_points;
+
+public:
+    latency_tracer(int id, const std::string &name)
+        : id(id),
+          name(name),
+          start_time(dsn_now_ns()),
+          previous_time(start_time),
+          request_type("default"){};
+
+    // this method is called for any other method which will be record the ts and time used
+    //
+    // name: generally, it is the method name that call this method. but you can define the more
+    // significant name to show the events of the moment
+    //
+    // current_time: current timestamp
+    //
+    // key_point: sometime, you may need know the total time used of some key point, if pass true,
+    // the point information will be record in map "key_points". default is false, means only focus
+    // the time used between current point and previous point
+    //
+    // end_point: if your trace is finised, you should record it
+    void add_point(const std::string &name,
+                   uint64_t current_time,
+                   bool key_point = false,
+                   bool end_point = false)
+    {
+        if (end_point) {
+            end_time = current_time;
+        }
+        if (key_point) {
+            key_points.emplace(name, current_time - start_time);
+        }
+        points.emplace(fmt::format("ts={}, {}", current_time, name), current_time - previous_time);
```

Comment:
> // record the the time used between current point and start point
> std::unordered_map<std::string, uint64_t> points;
> // record the the time used between current point and previous point
> std::unordered_map<std::string, uint64_t> key_points;

It seems that points also record  time used between current point and previous point, besides, I still can't understand the difference between key_points and points, could you please explain it to me? key_points record important points, why it only records current_time - previous_point? and points should record current_time - start_time?